### DISCUSSION TITLE
How do you test that context (de)hydration works?

### DISCUSSION THREAD

**Discussion by denjaland** [laravel/framework#53595] [Q&A]
Hi,

I just refactored our old (untested) code that used to consume `Queue::createPayloadUsing()` directly to start using Laravel's `Log\Context` (Still a bit frustrated it's written as part of the Log domain; I would have put it under the Queue domain myself, as now it feels as if it can only be used for logging whereas we use it to share context that we use in the async processes...)

Anyway.  As I said, we didn't test the code before, and I wanted to fix that now, but I'm not sure how I can test the fact that (de)hydation works as by default, our queue driver is set to sync.

This is an _attempt_, but obviously doesn't work:

```php
// Theme pushes context so it's shared in async job processing as well
Theme::set('blue');
// Simulating pushed a job to the queue
Queue::push(new TestQueueableJob());

Theme::set('red');
expect(Theme::get())->toBe('red');  // Success

// I would have hoped this could trigger the context hydration
Queue::pop();

expect(Theme::get())->toBe('blue');  // Fail: the value is still red
```

For your reference, in our service provider we are (de)hydrating the theme.  It does work as expected (emails dispatched from an async processed job, gets the right theme applied for instance); I'm just really unsure how I can test this propertly.

Basically, what I'm trying to test is:
1/ we push something to the queue
2/ we change the theme
3/ we process the job, and this job should actually still use the theme that was active when we pushed the job to the queue.
